### PR TITLE
Przezdziecki/404page

### PR DESCRIPTION
**Pull request by mat-px**
-Added a rewrite in firebase.json that will redirect a 404 to a new 404page.html
-Created 404page.html that just says that the page doesn't exist
-Not sure why there were spaces added/removed, I only viewed those, not edited

Just a note, first time doing anything with html and css, so hopefully it isn't too bad. I could create a css class for margins if you didn't want formatting in the html

### CODE REVIEW

File: .firebaserc [hunk 1]
```diff
@@ -18,4 +18,4 @@
       }
     }
   }
-}
\ No newline at end of file
+}
```
Comment (joe-studer-18) [inline .firebaserc:21]: Let's avoid changes in untouched files if possible, can you revert this one?

File: active/index.html [hunk 2]
```diff
@@ -230,8 +230,8 @@ <h4>Excellence</h4>
 
         <section id="about">
             <div class="container-fluid">
-                <iframe width="853" height="480" 
-                  src="https://my.matterport.com/show/?m=cjA8963SkH1" 
+                <iframe width="853" height="480"
```
Comment (joe-studer-18) [inline active/index.html:233]: Same comment as above, can you remove the changes to this file?

File: active/style/main.css [hunk 3]
```diff
@@ -6321,10 +6321,9 @@ body #values .img-responsive {
   		margin-left: auto;
 		margin-right: auto;
 	}
-	
 	body #values .center{
 		text-align: center;
-
```
Comment (joe-studer-18) [inline active/style/main.css:6327]: Same as before

File: active/views/values.html [hunk 4]
```diff
@@ -11,7 +11,7 @@
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.5.0/css/font-awesome.min.css">
     <link rel="stylesheet" href="style/main.css">
     <link rel="stylesheet" href="style/skin.css">
-    <link rel="stylesheet" href="style/values.css">    
+    <link rel="stylesheet" href="style/values.css">
```
Comment (joe-studer-18) [inline active/views/values.html:14]: See above

File: firebase.json [hunk 5]
```diff
@@ -10,6 +10,12 @@
       "public/",
       "active/img/originals"
     ],
+    "rewrites": [
```
Comment (joe-studer-18) [inline firebase.json:13]: Nit: can you add a comment describing the purpose of this rewrite?

File: active/404page.html [hunk 6]
```diff
@@ -0,0 +1,204 @@
+<!DOCTYPE html>
```
Comment (joe-studer-18) [inline active/404page.html:1]: Can you run 
```
firebase deploy --only hosting:devel
```

from this branch so I can check this file out live?

File: firebase.json [hunk 7]
```diff
@@ -10,6 +10,13 @@
...
+    /* Rewrites the URL to 404page.html if page doesn't exist */
+    "rewrites": [
+      {
+        "source": "**",
+        "destination": "404page.html"
+      }
+    ],
     "contributors": [
       "Joe Studer <joe.studer.18@gmail.com>",
       "Jared Hanisch <jared.hanisch@gmail.com>"
```
Comment (joe-studer-18) [inline firebase.json:22]: Go ahead and add yourself to contributors here too :)

File: .firebaserc [hunk 8]
```diff
@@ -18,4 +18,4 @@
       }
     }
   }
-}
```
Comment (joe-studer-18) [inline .firebaserc:21]: These file edits are still showing up btw

File: active/404page.html [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+<html>
```
Comment (joe-studer-18) [inline active/404page.html:2]: This page looks great!!!

File: firebase.json [hunk 10]
```diff
@@ -10,6 +10,13 @@
       "public/",
       "active/img/originals"
     ],
+    /* Rewrites the URL to 404page.html if page doesn't exist */
+    "rewrites": [
```
Comment (joe-studer-18) [inline firebase.json:14]: Not ideal, but can you copy this section to the staging and devel fields as well? That way the redirects will work the same regardless of which instance we use.

**Review by joe-studer-18 — CHANGES_REQUESTED** [review]

**Comment by mat-px** [comment]
Git is so weird and annoying, I have no idea why it wanted me to merge or why it added some cached folder.

**Review by joe-studer-18 — COMMENTED** [review]
Yeah it can be a real pain, it gets better with time though. There are still a couple of files with unnecessary edits - you can see which ones they are by heading to the "files changed" tab. There should only be two - 404page.html and firebase.json

**Review by joe-studer-18 — APPROVED** [review]

**Pull request merged**